neo4j/_sync/io/_pool.py

```diff
@@ -348,6 +348,9 @@ def acquire(
             self.address, deadline, liveness_check_timeout
         )
 
+    def is_direct(self):
```

Comment:
I like this method.

 * I think the base Pool class should have it too and should either `raise NotImplementedError` or become a `abc.ABC` (abstract base class).
 * There are already places in the driver that check whether the pool is direct or not (currently using `isinstance(pool, Neo4jPool)` or similar) . They could also be adopted to use this method.
 * It could also be an attribute just because the lookup might be ever so slightly faster than a method call. But that might be a bit much. Yeah whatever, that one was just a quick thought.